app/models/screening.rb

```diff
@@ -3,7 +3,18 @@
 class Screening < ApplicationRecord
   belongs_to :movie
   belongs_to :hall
+  has_many :reservations
   validates_associated :movie, :hall
   validates :start_time, :end_time, :price, presence: true
   validates :price, numericality: { greater_than: 0 }
+
+  def available_seats
+    taken_seats = []
+    reservations.each do |r|
+      r.tickets.each do |t|
```

Comment:
better names could be used here which would help when you are working with teams